src/main/java/seedu/address/model/appointment/AppointmentContainsDatePredicate.java

```diff
@@ -21,16 +22,20 @@ public AppointmentContainsDatePredicate(LocalDate date) {
 
     @Override
     public boolean test(Person person) {
+        requireNonNull(person);
         if (person.getAppointment() == null) {
             return false;
         } else if (this.date == null) {
             return true;
         }
 
-        LocalDate startDate = person.getAppointment().getStart().toLocalDate();
-        LocalDate endDate = person.getAppointment().getEnd().toLocalDate();
+        LocalDate startDate = person.getAppointmentStart().toLocalDate();
+        LocalDate endDate = person.getAppointmentEnd().toLocalDate();
```

Comment:
Nice! This follows the Law of Demeter. I shall use the new person methods in my own code when doing up the UI later